bugsnag/breadcrumbs.py

```diff
@@ -40,3 +60,47 @@ def to_dict(self) -> Dict[str, Union[str, FilterDict]]:
             'type': self.type.value,
             'metaData': FilterDict(self.metadata)
         }
+
+
+class Breadcrumbs:
+    def __init__(self, max_breadcrumbs: int):
+        self._max_breadcrumbs = max_breadcrumbs
+
+        # Calling resize is important for tests as we make many Breadcrumbs
+        # instances but they all have to share a ContextVar, so the size can
+        # leak between tests
+        self.resize(max_breadcrumbs)
+
+    def append(self, breadcrumb: Breadcrumb) -> None:
+        self._breadcrumbs.append(breadcrumb)
+
+    # Resize the list of breadcrumbs if configuration.max_breadcrumbs changes
+    def resize(self, new_max: int) -> None:
+        old_breadcrumbs = self._breadcrumbs
+        new_breadcrumbs = deque(old_breadcrumbs, maxlen=new_max)
+
+        _breadcrumbs.set(new_breadcrumbs)
+        self._max_breadcrumbs = new_max
+
+    # Create a copy of the current list of breadcrumbs for this context
+    def create_copy_for_context(self) -> None:
+        # Resizing will create a new deque and store it in the ContextVar,
+        # which will give the current context a new copy of the list
+        self.resize(self._max_breadcrumbs)
+
+    def to_list(self) -> List[Breadcrumb]:
```

Comment:
This class could instead implement `__iter__` and `__next__`, if you want `list(breadcrumbs)` to work

Reply:
Yeah that's more than reasonable. I figured you might have gone down that route already but if not, I'd check. :+1: